kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ResourceSelection.vue

```diff
@@ -271,7 +273,17 @@
       isSelectAllChecked() {
         // Returns true if all the resources in the topic are in the working resource pool
         const workingResourceIds = this.workingResourcePool.map(wr => wr.id);
-        return this.contentList.every(content => workingResourceIds.includes(content.id));
+
+        if (this.contentList.every(content => workingResourceIds.includes(content.id))) {
+          this.contentList.forEach(resource => {
+            if (resource.kind === ContentNodeKinds.TOPIC) {
+              this.addToWorkingResourcePool(resource.children.results);
+            }
+          });
+          return true;
+        } else {
+          return false;
+        }
```

Comment:
Lines 278-282 here seem out of place. Since `isSelectAllChecked` is supposed to just be a Boolean computed property, it can get really hairy if we introduce a side effect here like adding items to the resource pool.

This logic probably belongs below in the method we're passing as the handler function for the `@changeselectall` event on the `ContentCardList`. I'll add some more thoughts there.